test/src/main/java/org/apache/accumulo/test/fate/MultipleStoresIT.java

```diff
@@ -0,0 +1,553 @@
+    }
+  }
+
+  @Test
+  public void testMultipleFateInstances() throws Exception {
+    testMultipleFateInstances(FateInstanceType.META);
+    testMultipleFateInstances(FateInstanceType.USER);
+  }
+
+  private void testMultipleFateInstances(FateInstanceType storeType) throws Exception {
+    final String tableName = getUniqueNames(1)[0];
+    final int numFateIds = 500;
+    final boolean isUserStore = storeType.equals(FateInstanceType.USER);
+    final Set<FateId> allIds = new HashSet<>();
+    final FateStore<SleepingTestEnv> store1, store2;
+    final SleepingTestEnv testEnv1 = new SleepingTestEnv(50);
+    final SleepingTestEnv testEnv2 = new SleepingTestEnv(50);
+    final ZooUtil.LockID lock1 = new ZooUtil.LockID("/locks", "L1", 50);
+    final ZooUtil.LockID lock2 = new ZooUtil.LockID("/locks", "L2", 52);
+    final Set<ZooUtil.LockID> liveLocks = new HashSet<>();
+    final Predicate<ZooUtil.LockID> isLockHeld = liveLocks::contains;
+
+    if (isUserStore) {
+      createFateTable(client, tableName);
+      store1 = new UserFateStore<>(client, tableName, lock1, isLockHeld);
+      store2 = new UserFateStore<>(client, tableName, lock2, isLockHeld);
+    } else {
+      store1 = new MetaFateStore<>(FATE_DIR, zk, lock1, isLockHeld);
+      store2 = new MetaFateStore<>(FATE_DIR, zk, lock2, isLockHeld);
+    }
+    liveLocks.add(lock1);
+    liveLocks.add(lock2);
+
+    Fate<SleepingTestEnv> fate1 =
+        new Fate<>(testEnv1, store1, true, Object::toString, DefaultConfiguration.getInstance());
+    Fate<SleepingTestEnv> fate2 =
+        new Fate<>(testEnv2, store2, false, Object::toString, DefaultConfiguration.getInstance());
+
+    for (int i = 0; i < numFateIds; i++) {
+      FateId fateId;
+      // Start half the txns using fate1, and the other half using fate2
+      if (i % 2 == 0) {
+        fateId = fate1.startTransaction();
+        fate1.seedTransaction("op" + i, fateId, new SleepingTestRepo(), true, "test");
+      } else {
+        fateId = fate2.startTransaction();
+        fate2.seedTransaction("op" + i, fateId, new SleepingTestRepo(), true, "test");
+      }
+      allIds.add(fateId);
+    }
+    assertEquals(numFateIds, allIds.size());
+
+    // Should be able to wait for completion on any fate instance
+    for (FateId fateId : allIds) {
+      fate2.waitForCompletion(fateId);
+    }
+    // Ensure that all txns have been executed and have only been executed once
+    assertTrue(Collections.disjoint(testEnv1.executedOps, testEnv2.executedOps));
+    assertEquals(allIds, Sets.union(testEnv1.executedOps, testEnv2.executedOps));
+
+    fate1.shutdown(1, TimeUnit.MINUTES);
+    fate2.shutdown(1, TimeUnit.MINUTES);
+  }
+
+  @Test
+  public void testDeadReservationsCleanup() throws Exception {
+    testDeadReservationsCleanup(FateInstanceType.META);
+    testDeadReservationsCleanup(FateInstanceType.USER);
+  }
+
+  private void testDeadReservationsCleanup(FateInstanceType storeType) throws Exception {
```

Comment:
This test looks good.  Looking over this test realized we need another test in a follow on PR. Need a test that checks that when the reservation does not exists in the persisted store that every update fails.  Test would look something like the following.

```
    FateStore.FateTxStore<?> txStore = store1.reserve(fateId);
    // TODO delete the reservation in persisted store
    assertThrows(Exception.class, ()->txStore.push(repo));
    assertThrows(Exception.class, ()->txStore.setStatus(ReadOnlyFateStore.TStatus.IN_PROGRESS));
    assertThrows(Exception.class, ()->txStore.setTransactionInfo(t,v));
    //etc test all methods that write
```

For now may only be able to make this work with the UserFateStore as the current zookeeper schema does not support the conditional checks of reservations on updates.